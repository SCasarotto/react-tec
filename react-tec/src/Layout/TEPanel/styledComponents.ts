import styled from 'styled-components'

export const Container = styled.div<{ size?: string }>`
	position: relative;
	background-color: ${(props) => props.theme.white};
	border-radius: 5px;
	box-shadow: ${(props) => props.theme.shadowCenter};

	padding: 30px;
	margin-bottom: 30px;

	@media (max-width: 800px) {
		padding: 25px;
		margin-bottom: 25px;
	}
	@media (max-width: 650px) {
		padding: 20px;
		margin-bottom: 20px;
	}
	@media (max-width: 450px) {
		padding: 15px;
		margin-bottom: 15px;
		border-radius: 3px;
	}
	${(props) => {
		const { size } = props

		switch (size) {
			case 'full':
				return `width: 100%;`
			case 'three-quarter':
				return `
                    width: calc(75% - 20px);
                    @media (max-width: 700px) {
                        width: 100%;
                    }
                `
			case 'two-third':
				return `
                    width: calc(66.666% - 20px);
                    @media (max-width: 700px) {
                        width: 100%;
                    }
                `
			case 'half':
				return `
                    width: calc(50% - 15px);
                    @media (max-width: 700px) {
                        width: 100%;
                    }
                `
			case 'third':
				return `
                    width: calc(33.333% - 10px);
                    @media (max-width: 700px) {
                        width: 100%;
                    }
                `
			case 'quarter':
				return `
                    width: calc(25% - 10px);
                    @media (max-width: 700px) {
                        width: 100%;
                    }
                `
			case 'condensed':
				return `
                    width: auto;
                    @media (max-width: 700px) {
                        width: 100%;
                    }
                `
			default:
				return `width: 100%;`
		}
	}}
`
