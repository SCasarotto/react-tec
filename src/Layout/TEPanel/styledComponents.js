import styled from 'styled-components'

export const Container = styled.div`
	${(props) => {
		const { theme, size } = props

		let styles = `
            position: relative;
            background-color: ${theme.white};
            border-radius: 5px;
            box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 6px 0px, rgba(0, 0, 0, 0.05) 0px 0px 2px 0px;

            padding: 30px;
            margin-bottom: 30px;

            @media (max-width: 800px) {
                padding: 25px;
                margin-bottom: 25px;
            }
            @media (max-width: 650px) {
                padding: 20px;
                margin-bottom: 20px;
            }
            @media (max-width: 450px) {
                padding: 15px;
                margin-bottom: 15px;
                border-radius: 3px;
            }
        `

		switch (size) {
			case 'full':
				styles += `width: 100%;`
				break
			case 'two-third':
				styles += `
                    width: calc(66.666% - 20px);
                    @media (max-width: 700px) {
                        width: 100%;
                    }
                `
				break
			case 'half':
				styles += `
                    width: calc(50% - 15px);
                    @media (max-width: 700px) {
                        width: 100%;
                    }
                `
				break
			case 'third':
				styles += `
                    width: calc(33.333% - 10px);
                    @media (max-width: 700px) {
                        width: 100%;
                    }
                `
				break
			default:
				styles += `width: 100%;`
				break
		}

		return styles
	}}
`
