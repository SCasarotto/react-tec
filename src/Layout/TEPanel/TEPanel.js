//
//TE Version 0.3.0
//

import React from 'react'
import PropTypes from 'prop-types'

import TEPanelTitle from './../TEPanelTitle'

import { Container } from './styledComponents'

const TEPanel = (props) => {
	const {
		size,
		className,

		title,
		leftComponent,
		subtitle,
		rightComponent,

		children,
	} = props
	return (
		<Container className={className} size={size}>
			{(title || leftComponent || subtitle || rightComponent) && (
				<TEPanelTitle
					title={title}
					leftComponent={leftComponent}
					subtitle={subtitle}
					rightComponent={rightComponent}
					className='TEPanelTitleBar'
				/>
			)}
			{children}
		</Container>
	)
}

TEPanel.propTypes = {
	size: PropTypes.string,
}

TEPanel.defaultProps = {
	size: 'full',
}

export default TEPanel
