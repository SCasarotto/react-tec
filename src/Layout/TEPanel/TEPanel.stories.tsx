import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TEPanel, TEPanelProps } from './TEPanel'

export default {
	title: 'Layout/TEPanel',
	component: TEPanel,
	argTypes: {},
} as Meta

const Template: Story<TEPanelProps> = (args) => <TEPanel {...args} />

export const Default = Template.bind({})
Default.args = {}
