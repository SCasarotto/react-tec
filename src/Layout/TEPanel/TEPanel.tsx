import React from 'react'

import { TEPanelTitle, TEPanelTitleProps } from '../TEPanelTitle'

import { Container } from './styledComponents'

export type PanelSize =
	| 'full'
	| 'three-quarter'
	| 'two-third'
	| 'half'
	| 'third'
	| 'quarter'
	| 'condensed'
export interface TEPanelProps extends TEPanelTitleProps {
	size?: PanelSize
}
/**
 *
 * `TEPanel` simple containing element containing a `TEPanelTitle`. Is expected to be used as a child of `TEPanelWrapper`. In addition to the props below, view `TEPanelTitle` for additional props.
 *
 */
export const TEPanel: React.FC<TEPanelProps> = (props) => {
	const {
		size = 'full',
		className = '',
		title,
		leftComponent,
		subtitle,
		rightComponent,
		children,
	} = props
	return (
		<Container className={`TEPanel ${className}`} size={size}>
			{(title || leftComponent || subtitle || rightComponent) && (
				<TEPanelTitle
					title={title}
					leftComponent={leftComponent}
					subtitle={subtitle}
					rightComponent={rightComponent}
					className='TEPanelTitleBar'
				/>
			)}
			{children}
		</Container>
	)
}
