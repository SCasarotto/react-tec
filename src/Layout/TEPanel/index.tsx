import React from 'react'

import { TEPanelTitle, TEPanelTitleProps } from '../TEPanelTitle'

import { Container } from './styledComponents'

//Extending div handles className and Children but feels weird here because I don't spead the remained of the props
export interface TEPanelProps extends TEPanelTitleProps {
	//TODO: Fix this. styled-components is weird with refs
	ref?: any
	size?: string
}
export const TEPanel: React.FC<TEPanelProps> = (props) => {
	const {
		size = 'full',
		className = '',
		title,
		leftComponent,
		subtitle,
		rightComponent,
		children,
	} = props
	return (
		<Container className={`TEPanel ${className}`} size={size}>
			{(title || leftComponent || subtitle || rightComponent) && (
				<TEPanelTitle
					title={title}
					leftComponent={leftComponent}
					subtitle={subtitle}
					rightComponent={rightComponent}
					className="TEPanelTitleBar"
				/>
			)}
			{children}
		</Container>
	)
}
