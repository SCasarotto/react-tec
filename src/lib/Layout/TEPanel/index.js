import TEPanel from './TEPanel'

export default TEPanel
