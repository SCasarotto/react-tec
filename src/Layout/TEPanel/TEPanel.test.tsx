import React from 'react'
import { render } from '@testing-library/react'
import { TEPanel } from './TEPanel'

describe('TEPanel', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(<TEPanel />)
		expect(asFragment()).toMatchSnapshot()
	})
})
