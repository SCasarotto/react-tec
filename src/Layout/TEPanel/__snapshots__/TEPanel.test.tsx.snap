// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEPanel matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  border-radius: 5px;
  padding: 30px;
  margin-bottom: 30px;
  width: 100%;
}

@media (max-width:800px) {
  .c0 {
    padding: 25px;
    margin-bottom: 25px;
  }
}

@media (max-width:650px) {
  .c0 {
    padding: 20px;
    margin-bottom: 20px;
  }
}

@media (max-width:450px) {
  .c0 {
    padding: 15px;
    margin-bottom: 15px;
    border-radius: 3px;
  }
}

<div
    class="c0 TEPanel "
  />
</DocumentFragment>
`;
